persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/RocksDBStore.java

```diff
@@ -134,6 +137,13 @@
          try {
             db = handler.open(getLocation(), dataDbOptions());
             expiredDb = openDatabase(getExpirationLocation(), expiredDbOptions());
+
+            MetadataImpl existingMeta = handler.loadMetadata();
+            if (existingMeta != null) {
+               // TODO Perform validation/migration
```

Comment:
Do we have a JIRA for this?

Reply:
Maybe for this iteration we can just verify the version and if there is a mismatch log a WARN/INFO message?

Reply:
Tbh as this is the first version that has this logic, I'm not sure how there could be a mismatch. It's also possible (in fact desirable) in future versions that loading from a previous major should not require migration, so logging would only add noise IMO.